tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +103,225 @@ def GetFieldsForTypeName(
+    )
+    new_match = Match(
+        field_list,
+        entity_type,
+        match_score
+    )
+    return new_match
 
   def GetEntityTypesFromFields(self,
-                               field_list: List[EntityTypeField],
-                               general_type: str = None) -> List[EntityType]:
-    """Get a list of EntityType objects matching a list of EntityTypeFields.
+                               field_list: List[StandardField],
+                               return_size: int = 0,
+                               general_type: str = None) -> List[Match]:
+    """Get a list of Match objects containg information on a strength of a
+    match between all entity types defined in DBO and a list of concrete
+    fields.
 
     Args:
-        field_list: a list of EntityTypeField objects to match to an entity
-        general_type: a string indicating a general type name to filter return
-          results.
+      field_list: A list of StandardField objects to match to an entity.
+      general_type: A string indicating a general type name to filter return
+        results.
+      return_size: An int for the length of the return list of matches.
+        e.g. if return_size is 10, the 10 matches with the highest score will
+        be returned.
+    Returns:
+      A sorted list of Match objects.
+    """
+    entity_type_list = []
+    type_namespaces_list = self.universe.GetEntityTypeNamespaces()
+    for tns in type_namespaces_list:
+      for entity_type in tns.valid_types_map.values():
+        if entity_type.is_abstract or entity_type.GetAllFields() == {}:
+          continue
+        if general_type is not None:
+          if general_type.upper() in entity_type.unqualified_parent_names:
+            entity_type_list.append(entity_type)
+        else:
+          entity_type_list.append(entity_type)
+
+    match_list = []
+    for entity_type in entity_type_list:
+      match_list.append(self._CreateMatch(field_list, entity_type))
+
+    match_list_sorted = sorted(
+        match_list,
+        key=lambda x: x.GetMatchScore(),
+        reverse=True
+    )
+    if return_size > 0:
+      return match_list_sorted[:return_size]
+    return match_list_sorted
+
+  def PrintFieldSetComparison(self, match: Match)-> str:
+    """creates a text representation of field set relations for a given match.
 
+    Takes the intersection and differences in sets between a set of fields
+    belonging to an entity type and concrete entity to create a big string
+    representation of matching fields within a match. This method will be called
+    by app.py when a user wants to visualize the field relationships between a
+    list of fields and an entity type within a match.
+
+    Args:
+      match: An instance of Match class for which a field set relation wants
+      to be visualized.
     Returns:
-          entities: a list of EntityType objects matching the provided list of
-          fields.
+      A string to visualize a field set relation
```

Comment:
can you give an example here please?

Reply:
![image](https://user-images.githubusercontent.com/2355898/145880093-a5e39c84-1b18-4998-b909-6e30dc7212a5.png)

Reply:
This is what it looks like.